### ISSUE TITLE
VARIANT data type in Databricks not implemented

### ISSUE THREAD

**Issue by KenHBS** [sqlfluff/sqlfluff#6120]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

When trying to create a table with the data type `VARIANT`, an unparsable section is detected. When I replace the `VARIANT` data type, everything works as expected

Example query, using version 3.1.0:
```
CREATE TABLE my_table (
    field_a STRING,
    field_b VARIANT
);

```
The `VARIANT` data type is specific to the Databricks dialect and was introduced in June 2024. It doesn't exist in SparkSQL nor ANSI SQL. Perhaps this could be as easy as adding another unreserved keyword to `sqlfluff/dialects/dialect_databricks_keywords.py`?

### Expected Behaviour

When running

```bash
sqlfluff lint test.sql
```
I expect the `All Finished 📜 🎉! ` to appear, without an error pointing to an unparsable section

### Observed Behaviour

```
....                                                                                                                                                  
L:   1 | P:  23 |  PRS | Line 1, Position 23: Found unparsable section: '(\n   
                       | field_a STRING,\n    field_b VARIAN...'
....
```
When parsing the file with `sqlfluff parse test.sql`, I receive:
```
[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    statement:
[L:  1, P:  1]      |        create_table_statement:
[L:  1, P:  1]      |            keyword:                                          'CREATE'
[L:  1, P:  7]      |            whitespace:                                       ' '
[L:  1, P:  8]      |            keyword:                                          'TABLE'
[L:  1, P: 13]      |            whitespace:                                       ' '
[L:  1, P: 14]      |            table_reference:
[L:  1, P: 14]      |                naked_identifier:                             'my_table'
[L:  1, P: 22]      |            [META] indent:
[L:  1, P: 22]      |            [META] dedent:
[L:  1, P: 22]      |    whitespace:                                               ' '
[L:  1, P: 23]      |    unparsable:                                               !! Expected: 'Nothing else in FileSegment.'
[L:  1, P: 23]      |        start_bracket:                                        '('
[L:  1, P: 24]      |        newline:                                              '\n'
[L:  2, P:  1]      |        whitespace:                                           '    '
[L:  2, P:  5]      |        word:                                                 'field_a'
[L:  2, P: 12]      |        whitespace:                                           ' '
[L:  2, P: 13]      |        word:                                                 'STRING'
[L:  2, P: 19]      |        comma:                                                ','
[L:  2, P: 20]      |        newline:                                              '\n'
[L:  3, P:  1]      |        whitespace:                                           '    '
[L:  3, P:  5]      |        word:                                                 'field_b'
[L:  3, P: 12]      |        whitespace:                                           ' '
[L:  3, P: 13]      |        word:                                                 'VARIANT'
[L:  3, P: 20]      |        newline:                                              '\n'
[L:  4, P:  1]      |        end_bracket:                                          ')'
[L:  4, P:  2]      |        semicolon:                                            ';'
[L:  4, P:  3]      |    newline:                                                  '\n'
[L:  5, P:  1]      |    [META] end_of_file:

==== parsing violations ====
L:   1 | P:  23 |  PRS | Line 1, Position 23: Found unparsable section: '(\n    field_a
                       | STRING,\n    field_b VARIAN...'
```

### How to reproduce

Store the query above in `test.sql` and run `sqlfluff lint test.sql --dialect databricks`

### Dialect

The Databricks dialect

### Version

3.1.0

### Configuration

```
[tool.sqlfluff.core]
dialect = "databricks"
templater = "jinja"
max_line_length = 150
```

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
